### PR TITLE
[skip vbump] patch release 0.15.1

### PR DESCRIPTION
**Pull request by donyunardi**
Related to #1108

### CODE REVIEW

**Comment by donyunardi** [comment]
@insightsengineering/nest-core-dev 
Due to .Rd update in teal.reporter `main` branch (related to [this](https://github.com/insightsengineering/teal.reporter/pull/247) PR), the `man/reporter_previewer_module.Rd` got updated by the bot during [this](https://github.com/insightsengineering/teal/pull/1105/commits) PR as we fix the bug.

This automatic update broke the links in `reporter_previewer_module` documentation during CRAN check because it's comparing the link to latest release teal.reporter in CRAN.

I restored the man/reporter_previewer_module.Rd file and stop the bot from making further changes by adding `[skip roxygen]` to my commit message in this PR.

However, I'm worried that the next commit comes in to `teal` will trigger this automatic update again.
Does this mean we need to patch-release teal.reporter too? Any better idea?

**Comment by donyunardi** [comment]
nvmnd, `[skip roxygen]` doesn't work. =(
Besides, this doesn't feel like an optimal solution.

**Comment by donyunardi** [comment]
Another idea is to provide a direct link to the page in question. We can do this as temporary solution until teal.reporter is released to CRAN.

The roxygen note will look something like this:
```r
#' This function wraps [teal.reporter::reporter_previewer_ui()](https://insightsengineering.github.io/teal.reporter/latest-tag/reference/reporter_previewer_ui.html) and
#' [teal.reporter::reporter_previewer_srv()](https://insightsengineering.github.io/teal.reporter/latest-tag/reference/reporter_previewer_srv.html) into a `teal_module` to be
#' used in `teal` applications.
```

**Comment by kartikeyakirar** [comment]
For address the issue of automatic updates made by the CI,  we could revert changes made to the ` previewer.R` file until a patch release of `teal` is prepared. this will be more quicker solution

The documentation for `teal,` related to` teal.reporter,` needs to be updated in a manner that aligns with the simultaneous release of both packages.

> Does this mean we need to patch-release teal.reporter too? 

An approach involves updating the `teal `documentation for `teal.reporter` immediately and then releasing both as a patch. However, my concern is  documentation updates across the package `teal.reporter` should not lead to any issue as patch release.

**Comment by vedhav** [comment]
Ideally, we should do a patch release of `teal.reporter` and then the patch release of `teal`. Eventually, this change has to come to teal docs. And I don't feel reverting a valid commit just for the time being is a good solution.

**Comment by m7pr** [comment]
Im with @averissimo

File: NEWS.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-# teal 0.15.0.9005
+# teal 0.15.1
+
+### Bug fixes
+* Fixed a bug that stopped the `module_specific` feature from working when a teal module has more than one `datanames`.
```
Comment (chlebowa) [inline NEWS.md:4]:
```suggestion
* Fixed a bug that stopped the `module_specific` feature when a `teal` module specified `datanames` of length more than 1.
```

File: NEWS.md [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
+
+### Miscellaneous
+* Update links in vignettes
```
Comment (chlebowa) [inline NEWS.md:7]:
```suggestion
* Updated links in vignettes
```

**Comment by chlebowa** [comment]
So this is because we decided to omit explicit `latest-tag` from URLs? Do I have this correct?

**Comment by donyunardi** [comment]
> So this is because we decided to omit explicit latest-tag from URLs? Do I have this correct?

I believe this is happening because the bot is updating/creating `reporter_previewer_module.Rd` (which contains links to teal.reporter functions) using the `main` teal.reporter instead of the latest release or CRAN release of teal.reporter. It's the R environment and package versions that run during the CICD process.

**Comment by donyunardi** [comment]
You can see this locally and compare the result between:
1. Install teal.reporter 0.3.0 from CRAN and build the teal document
2. Install teal.reporter 0.3.0.9001 from pharmaverse/github and build the teal document

**Comment by gogonzo** [comment]
> You can see this locally and compare the result between:
> 
> 1. Install teal.reporter 0.3.0 from CRAN and build the teal document
> 2. Install teal.reporter 0.3.0.9001 from pharmaverse/github and build the teal document

@donyunardi current PR still refers to the teal.reporter 0.3.0.9001. This will not go through the CRAN checks. 

Refering to the yesterdays talk we have several way to solve this problem if we want to re-release `teal` now:

1. To release (now) patch version of `teal.reporter` and then `teal`
2. To release `teal` with documentation link to the CRAN released `teal.reporter`. To do so we will need to force-push `teal` to the main branch and fix the automation issue after merge/tagging.
  - Sooner or later updated `teal.reporter` will be released and then we will HAVE TO release `teal` also because of reverse dependency failure.
 3. We revert `teal.reporter` documentation enhancement in sake to release now `teal` and then re-revert. 
 
 I personally support (3)

**Comment by vedhav** [comment]
Since most of us prefer to revert. This the PR https://github.com/insightsengineering/teal.reporter/pull/254 reverts the doc changes related to the issue. We can revert it back after teal is released.

**Comment by donyunardi** [comment]
Submitted to CRAN.
Awaiting feedback.

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @donyunardi

**Pull request merged**